src/test/java/org/threeten/extra/chrono/TestInternationalFixedChronology.java

```diff
@@ -0,0 +1,1306 @@
+            {2012, -1, -1, 2013, 1, 1, MONTHS, 7},
+            {2012, -1, -1, 2011, 13, 28, MONTHS, -6},
+            {2012, -1, -1, 2011, 0, 0, MONTHS, -6},
+            {2012, -1, -1, 2012, 1, 1, MONTHS, -5},
+            {2012, -1, -1, 2016, -1, -1, WEEKS, 52 * 4},
+            {2012, 0, 0, 2012, 6, 28, MONTHS, -7},
+            {2012, 0, 0, 2012, -1, -1, MONTHS, -6},
+            {2012, 0, 0, 2012, 7, 1, MONTHS, -6},
+            {2011, 0, 0, 2012, 6, 28, MONTHS, 5},
+            {2011, 0, 0, 2012, -1, -1, MONTHS, 6},
+            {2011, 0, 0, 2012, 7, 1, MONTHS, 6},
+        };
+    }
+
+    @DataProvider(name = "until_period")
+    Object[][] data_until_period() {
+        return new Object[][] {
+            {2014, 5, 26, 2014, 5, 26, 0, 0, 0},
+            {2014, 5, 26, 2014, 6, 4, 0, 0, 6},
+            {2014, 5, 26, 2014, 5, 20, 0, 0, -6},
+            {2014, 5, 26, 2014, 6, 5, 0, 0, 7},
+            {2014, 5, 26, 2014, 6, 25, 0, 0, 27},
+            {2014, 5, 26, 2014, 6, 26, 0, 1, 0},
+            {2014, 5, 26, 2015, 5, 25, 0, 12, 27},
+            {2014, 5, 26, 2015, 5, 26, 1, 0, 0},
+            {2014, 5, 26, 2024, 5, 25, 9, 12, 27},
+
+            {2011, 13, 26, 2013, 13, 26, 2, 0, 0},
+            {2011, 13, 26, 2012, 13, 26, 1, 0, 0},
+            {2012, 13, 26, 2011, 13, 26, -1, 0, 0},
+            {2012, 13, 26, 2013, 13, 26, 1, 0, 0},
+
+            {2011, 13, 6, 2012, 13, 6, 1, 0, 0},
+            {2012, 13, 6, 2011, 13, 6, -1, 0, 0},
+            {2011, 13, 1, 2012, 13, 7, 1, 0, 6},
+            {2012, 13, 7, 2011, 13, 1, -1, 0, -6},
+
+            {2011, 12, 28, 2012, 13, 1, 1, 0, 1},
+            {2012, 13, 1, 2011, 12, 28, -1, 0, -1},
+            {2013, 13, 6, 2012, 13, 6, -1, 0, 0},
+            {2012, 13, 6, 2013, 13, 6, 1, 0, 0},
+
+            // start with Year Day
+            {2012, 0, 0, 2012, 0, 0, 0, 0, 0},
+            {2012, 0, 0, 2013, 0, 0, 1, 0, 0},
+            {2011, 0, 0, 2010, 0, 0, -1, 0, 0},
+            {2000, 0, 0, 2001, 0, 0, 1, 0, 0},
+            {2007, 0, 0, 2008, 1, 1, 0, 0, 1},
+            {2005, 0, 0, 2006, 2, 1, 0, 1, 1},
+            {2003, 0, 0, 2004, -1, -1, 0, 6, 1},
+            {2003, 0, 0, 2004, 7, 1, 0, 6, 2},
+            {2004, 0, 0, 2004, -1, -1, 0, -7, 0},
+            {2004, 0, 0, 2004, 7, 1, 0, -6, -27},
+            {2003, 0, 0, 2005, 1, 1, 1, 0, 1},
+            {2003, 0, 0, 2002, 13, 28, -1, 0, -1},
+
+            // start with one day before Year Day
+            {2003, 13, 28, 2004, -1, -1, 0, 6, 1},
+            {2003, 13, 28, 2004, 7, 1, 0, 6, 2},
+            {2004, 13, 28, 2004, -1, -1, 0, -7, 0},
+            {2004, 13, 28, 2004, 7, 1, 0, -6, -27},
+            {2003, 13, 28, 2005, 1, 1, 1, 0, 2},
+            {2003, 13, 28, 2002, 13, 28, -1, 0, 0},
+            {2007, 13, 28, 2008, 1, 1, 0, 0, 2},
+            {2005, 13, 28, 2006, 2, 1, 0, 1, 1},
+
+            // start with Leap Day
+            {2008, -1, -1, 2008, -1, -1, 0, 0, 0},
```

Comment:
You should probably add leap day -> leap day verification (that it returns just years).